tests/integration/test_provider.py

```diff
@@ -133,41 +140,73 @@ async def test_remove_application_removes_user_and_acls(ops_test: OpsTest, usern
 
 
 @pytest.mark.abort_on_fail
-async def test_change_client_topic(ops_test: OpsTest):
```

Comment:
I am not sure of the behavior of the lib so hopefully my point is still applicable. 
But I think we should add a test to validate the behavior of the charm when we change a field for a related application. 

e.g `change topic`, or `change roles` etc.

Reply:
I believe in this case we just delete the relation and create a new one. We had a chat about this with @marceloneppel when working on the data-platform-libs, and noticed that there was no "CredentialChanged" event in the databases signature. The understanding is that the password/role/topic rotation will be done by removing relation and adding new relation. See the comments [here](https://github.com/canonical/data-platform-libs/pull/28#discussion_r1037105453). 

@delgod please double check the considerations here are accurate and correct

Reply:
If the charm user wants to make password/role/topic rotation, it should be done by removing the relation and adding new relation.
But if we (charm developers) want to make password/role/topic rotation, we can change the relation, but we should be sure that the consumer library has a special event for this. Right now, I think we don't have a `CredentialChanged` event.

Reply:
Nope, we don't. Let's then leave this aside for the time being, and rather thing a bit about how to implement this, such that charm developers don't overuse this, exposing to the final user this functionality